### PR TITLE
chore: add types for hyperswarm

### PR DESCRIPTION
**Pull request by achou11**
Towards #14 

Required some code changes that I'm not entirely sure about (although existing tests pass).

### CODE REVIEW

File: lib/discovery/index.js [hunk 1]
```diff
@@ -552,7 +554,7 @@ export class Discovery extends TypedEmitter {
       topic.mdns?.lookup(serviceType)
     }
 
-    if (dhtActive) {
+    if (dhtActive && this.dht && topic.dht) {
```
Comment (achou11) [inline lib/discovery/index.js:557]: probably the most unsure code change

File: types/modules.d.ts [hunk 2]
```diff
@@ -70,7 +70,130 @@ declare module 'time-ordered-set' {
...
+    banned: boolean
+    tried: boolean
+    explicit: boolean
+    queued: boolean
+    client: boolean
+    topics: Buffer[]
+    attempts: number
+    priority: number
+
+    constructor(opts: { publicKey: Buffer; relayAddresses: any[] })
```
Comment (achou11) [inline types/modules.d.ts:134]: slightly out of scope for this PR, but @sethvincent do you know what the type for the `relayAddresses` is?

File: lib/discovery/index.js [hunk 3]
```diff
@@ -1203,21 +1205,21 @@ export class Topic extends TypedEmitter {
    * @param {TopicServiceStatus} [options.dhtStatus]
    * @param {TopicServiceStatus} [options.mdnsStatus]
    * @param {MdnsDiscovery} [options.mdns]
-   * @param {Hyperswarm} [options.dht]
+   * @param {import('hyperswarm').PeerDiscoverySession} [options.dht]
```
Comment (achou11) [inline lib/discovery/index.js:1208]: based on usage, guessed that it was the discovery session we want to pass here, not the hyperswarm instance itself

File: lib/discovery/index.js (hunk 1 above)
Comment (sethvincent) [reply lib/discovery/index.js:557]: :+1: i think this makes sense

File: lib/discovery/index.js (hunk 3 above)
Comment (sethvincent) [reply lib/discovery/index.js:1208]: ah yeah, that looks right

File: types/modules.d.ts (hunk 2 above)
Comment (sethvincent) [reply types/modules.d.ts:134]: it looks like an array of objects with `host` and `port` properties based on hyperswarm/dht: https://github.com/hyperswarm/dht/blob/bb21a0191c1d96a727dcf82ffe3cb8967fb7c474/test/messages.js#L258

**Review by sethvincent — APPROVED** [review]
:+1:

File: types/modules.d.ts (hunk 2 above)
Comment (achou11) [reply types/modules.d.ts:134]: ah nice! I'll update in a follow-up

**Pull request merged**